### ISSUE TITLE
Stream for RTSP camera never starts

### ISSUE THREAD

**Issue by blakeblackshear** [AlexxIT/go2rtc#221]
I am unable to get go2rtc to stream from my RTSP cameras from my development machine.

Environment:
- Ubuntu 22.04 on Framework Laptop

Steps to reproduce:

1. Download binary release
2. Run with
```
./go2rtc_linux_amd64 -config '{"streams": {"back": "rtsp://viewer:test1234@10.0.10.10:554/cam/realmonitor?channel=1&subtype=2"}, "log": {"level": "trace"}}'
``` 
3. Select `mp4` in the go2rtc UI

Here are the trace logs
```
10:29:36.744 INF go2rtc version 1.0.1 linux/amd64
10:29:36.744 TRC [api] register path path=/
10:29:36.744 TRC [api] register path path=/api
10:29:36.744 TRC [api] register path path=/api/config
10:29:36.744 TRC [api] register path path=/api/exit
10:29:36.744 TRC [api] register path path=/api/ws
10:29:36.744 INF [api] listen addr=:1984
10:29:36.744 INF [rtsp] listen addr=:8554
10:29:36.744 TRC [api] register path path=/api/devices
10:29:36.744 TRC [api] register path path=/static
10:29:36.744 TRC [api] register path path=/streams
10:29:36.744 TRC [api] register path path=/stream/
10:29:36.744 TRC [api] register path path=/stream
10:29:36.744 INF [webrtc] listen addr=:8555
10:29:36.744 TRC [api] register path path=/api/webrtc
10:29:36.744 TRC [api] register path path=/api/frame.mp4
10:29:36.744 TRC [api] register path path=/api/stream.mp4
10:29:36.744 TRC [api] register path path=/api/frame.jpeg
10:29:36.744 TRC [api] register path path=/api/stream.mjpeg
10:29:36.744 INF [srtp] listen addr=:8443
10:29:36.744 TRC [api] register path path=/api/homekit
10:29:36.744 TRC [api] register path path=/api/stack
10:30:43.190 TRC [api] GET /api/stream.mp4?src=back [::1]:60060
10:30:43.190 TRC [mp4] GET map[Accept:[text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9] Accept-Encoding:[gzip, deflate, br] Accept-Language:[en-US,en;q=0.9,fr;q=0.8] Cache-Control:[max-age=0] Connection:[keep-alive] Cookie:[csrftoken=iY5HnudvW41pvtXmkubjSNvNdRf7sDbQ7dHXXUZVplFnCG1JW7iBWDaTC7HnL7h9] Referer:[http://localhost:1984/] Sec-Ch-Ua:["Not_A Brand";v="99", "Google Chrome";v="109", "Chromium";v="109"] Sec-Ch-Ua-Mobile:[?0] Sec-Ch-Ua-Platform:["Linux"] Sec-Fetch-Dest:[document] Sec-Fetch-Mode:[navigate] Sec-Fetch-Site:[same-origin] Sec-Fetch-User:[?1] Upgrade-Insecure-Requests:[1] User-Agent:[Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36]]
10:30:43.230 TRC [api] GET /api/stream.mp4?src=back [::1]:60060
10:30:43.230 TRC [mp4] GET map[Accept:[*/*] Accept-Encoding:[identity;q=1, *;q=0] Accept-Language:[en-US,en;q=0.9,fr;q=0.8] Connection:[keep-alive] Cookie:[csrftoken=iY5HnudvW41pvtXmkubjSNvNdRf7sDbQ7dHXXUZVplFnCG1JW7iBWDaTC7HnL7h9] Range:[bytes=0-] Referer:[http://localhost:1984/api/stream.mp4?src=back] Sec-Ch-Ua:["Not_A Brand";v="99", "Google Chrome";v="109", "Chromium";v="109"] Sec-Ch-Ua-Mobile:[?0] Sec-Ch-Ua-Platform:["Linux"] Sec-Fetch-Dest:[video] Sec-Fetch-Mode:[no-cors] Sec-Fetch-Site:[same-origin] User-Agent:[Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36]]
10:30:43.230 TRC [streams] consumer=0 candidate=0 media="video, recvonly, 0 H264/0, 0 H265/0"
10:30:43.230 DBG [streams] probe producer url=rtsp://viewer:test1234@10.0.10.10:554/cam/realmonitor?channel=1&subtype=2
10:30:43.231 TRC [rtsp] client request:
DESCRIBE rtsp://10.0.10.10:554/cam/realmonitor?channel=1&subtype=2 RTSP/1.0
Accept: application/sdp
Require: www.onvif.org/ver20/backchannel
CSeq: 1


10:30:43.236 TRC [rtsp] client response:
RTSP/1.0 401 Unauthorized
Cseq: 1
Www-Authenticate: Digest realm="Login to 3HPAG9EF51D3145", nonce="244e6bc2c91e215626b13498a7aca068"


10:30:43.236 TRC [rtsp] client request:
DESCRIBE rtsp://10.0.10.10:554/cam/realmonitor?channel=1&subtype=2 RTSP/1.0
Accept: application/sdp
Require: www.onvif.org/ver20/backchannel
CSeq: 2
Authorization: Digest username="viewer", realm="Login to 3HPAG9EF51D3145", nonce="244e6bc2c91e215626b13498a7aca068", uri="/cam/realmonitor?channel=1&subtype=2", response="ea928d53fb2b822b67a690995b149c70"


10:30:43.270 TRC [rtsp] client response:
RTSP/1.0 200 OK
X-Accept-Dynamic-Rate: 1
Cache-Control: must-revalidate
Cseq: 2
Content-Length: 723
Content-Type: application/sdp
Content-Base: rtsp://10.0.10.10:554/cam/realmonitor?channel=1&subtype=2/

v=0
o=- 2252223832 2252223832 IN IP4 0.0.0.0
s=Media Server
c=IN IP4 0.0.0.0
t=0 0
a=control:*
a=packetization-supported:DH
a=rtppayload-supported:DH
a=range:npt=now-
m=video 0 RTP/AVP 96
a=control:trackID=0
a=framerate:5.000000
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1;profile-level-id=640028;sprop-parameter-sets=Z2QAKKwbGoB4AiflwFuAgICgAAB9AAAE4h0MAPoAAAvrwXeXGhgB9AAAF9eC7y4UAA==,aO44MAA=
a=recvonly
m=audio 0 RTP/AVP 97
a=control:trackID=1
a=rtpmap:97 MPEG4-GENERIC/16000
a=fmtp:97 streamtype=5;profile-level-id=1;mode=AAC-hbr;sizelength=13;indexlength=3;indexdeltalength=3;config=1408
a=recvonly
m=audio 0 RTP/AVP 97
a=control:trackID=5
a=rtpmap:97 MPEG4-GENERIC/16000
a=sendonly

10:30:43.270 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
10:30:43.270 TRC [streams] match producer:0:0 => consumer:0:0 codec="96 H264/90000"
10:30:43.271 TRC [rtsp] client request:
SETUP rtsp://10.0.10.10:554/cam/realmonitor?channel=1&subtype=2/trackID=0 RTSP/1.0
Transport: RTP/AVP/TCP;unicast;interleaved=0-1
CSeq: 3
Authorization: Digest username="viewer", realm="Login to 3HPAG9EF51D3145", nonce="244e6bc2c91e215626b13498a7aca068", uri="/cam/realmonitor?channel=1&subtype=2/trackID=0", response="90a76bea175383a54b8fef48c8151010"


10:30:47.150 TRC [rtsp] client request:
DESCRIBE rtsp://10.0.10.10:554/cam/realmonitor?channel=1&subtype=2 RTSP/1.0
Accept: application/sdp
CSeq: 4


10:30:47.155 TRC [rtsp] client response:
RTSP/1.0 401 Unauthorized
Cseq: 4
Www-Authenticate: Digest realm="Login to 3HPAG9EF51D3145", nonce="da8c14979863e657df6fd04877bb72f3"


10:30:47.155 TRC [rtsp] client request:
DESCRIBE rtsp://10.0.10.10:554/cam/realmonitor?channel=1&subtype=2 RTSP/1.0
Accept: application/sdp
CSeq: 5
Authorization: Digest username="viewer", realm="Login to 3HPAG9EF51D3145", nonce="da8c14979863e657df6fd04877bb72f3", uri="/cam/realmonitor?channel=1&subtype=2", response="503a3858b2dcbfb329579ee6b9a318a1"


10:30:47.189 TRC [rtsp] client response:
RTSP/1.0 200 OK
Cseq: 5
X-Accept-Dynamic-Rate: 1
Content-Type: application/sdp
Content-Length: 635
Content-Base: rtsp://10.0.10.10:554/cam/realmonitor?channel=1&subtype=2/
Cache-Control: must-revalidate

v=0
o=- 2252223832 2252223832 IN IP4 0.0.0.0
s=Media Server
c=IN IP4 0.0.0.0
t=0 0
a=control:*
a=packetization-supported:DH
a=rtppayload-supported:DH
a=range:npt=now-
m=video 0 RTP/AVP 96
a=control:trackID=0
a=framerate:5.000000
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1;profile-level-id=640028;sprop-parameter-sets=Z2QAKKwbGoB4AiflwFuAgICgAAB9AAAE4h0MAPoAAAvrwXeXGhgB9AAAF9eC7y4UAA==,aO44MAA=
a=recvonly
m=audio 0 RTP/AVP 97
a=control:trackID=1
a=rtpmap:97 MPEG4-GENERIC/16000
a=fmtp:97 streamtype=5;profile-level-id=1;mode=AAC-hbr;sizelength=13;indexlength=3;indexdeltalength=3;config=1408
a=recvonly
```

It just hangs here and the stream never plays in the browser.

Of note:

1. The same version of go2rtc is successfully running on my server inside frigate with the exact same config
2. Both of my cameras work with go2rtc in frigate, and neither work from my development machine
3. Both machines are hardwired into the same switch
4. The streams **do** work if I prepend `ffmpeg:` in front of the rtsp url

**Comment by blakeblackshear** [comment]
I think I figured this out. It does work if I stop Frigate on my server. So this happens only on the second go2rtc connection to the camera.

**Comment by blakeblackshear** [comment]
using this allows the second stream to connect as well:
```yaml
go2rtc:
  streams:
    back: rtsp://viewer:test1234@10.0.10.10:554/cam/realmonitor?channel=1&subtype=2#backchannel=0
```

**Comment by AlexxIT** [comment]
> So this happens only on the second go2rtc connection to the camera.

What you mean about second connection?

I can see describe after setup. This is very strange. I'll think why this may happens.

**Comment by blakeblackshear** [comment]
> What you mean about second connection?

Connecting to the camera again with go2rtc from a different computer.

**Comment by AlexxIT** [comment]
OK. This is problem with some Dahua cameras that fail setup command with backchannel.
I already fixed it, it will be in the next release.
In the future, I will not enable backchannel by default. This will reduce the number of such problems.

**Comment by AlexxIT** [comment]
Should be fixed in v1.1.0
